rust/nasl-interpreter/src/built_in_functions/array.rs

```diff
@@ -83,4 +144,119 @@ mod tests {
         assert_eq!(parser.next(), Some(Ok(make_dict!())));
         assert_eq!(parser.next(), Some(Ok(make_dict!())));
     }
+
+    #[test]
+    fn make_list() {
+        let code = r###"
+        a = [2,4]
+        make_list(1, 0);
+        make_list();
+        make_list(1,NULL,2);
+        b = make_array("el", 6);
+        make_list(1, 0, b);
+        make_list(1, 0, a);
+        "###;
+        let storage = DefaultSink::new(false);
+        let mut register = Register::default();
+        let loader = NoOpLoader::default();
+        let mut interpreter = Interpreter::new("1", &storage, &loader, &mut register);
+        let mut parser =
+            parse(code).map(|x| interpreter.resolve(&x.expect("no parse error expected")));
+        assert_eq!(parser.next(),
+                   Some(Ok(NaslValue::Array(vec![NaslValue::Number(2),NaslValue::Number(4)]))));
+        assert_eq!(parser.next(),
+                   Some(Ok(NaslValue::Array(vec![NaslValue::Number(1),NaslValue::Number(0)]))));
+        assert_eq!(parser.next(), Some(Ok(NaslValue::Array([].into()))));
+        assert_eq!(parser.next(),
+                   Some(Ok(NaslValue::Array(vec![NaslValue::Number(1),NaslValue::Number(2)]))));
+        parser.next();
+        assert_eq!(parser.next(),
+                   Some(Ok(NaslValue::Array(vec![
+                       NaslValue::Number(1),
+                       NaslValue::Number(0),
+                       NaslValue::Number(6)]))));
+        assert_eq!(parser.next(),
+                   Some(Ok(NaslValue::Array(vec![
+                       NaslValue::Number(1),
+                       NaslValue::Number(0),
+                       NaslValue::Number(2),
+                       NaslValue::Number(4)]))));
+    }
+
+    #[test]
+    fn sort() {
+        let code = r###"
+        a = make_array(5, 6, 7, 8);
```

Comment:
What is this first line for?

Reply:
This creates an array which is used for extending the list. 
The original nasl function, removes the keys and insert the values in the list. I just wanted to check that the array's values are sorted as well .